docs/proposals/27-project-group.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Project Groups
+gep-number: 27
+creation-date: 2024-02-25
+status: implementable
+authors:
+- "@tobschli"
+- "@oliver-goetz"
+reviewers:
+- "@rfranzke"
+- "@ScheererJ"
+---
+
+# GEP-27: Project Groups to enable Private Seeds
+
+## Table of Contents
+
+- [GEP-27: Project Groups to enable Private Seeds](#gep-27-project-groups-to-enable-private-seeds)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+      - [ProjectGroup](#projectgroup)
+      - [SeedBinding](#seedbinding)
+  - [Alternatives](#alternatives)
+    - [Implementation without ProjectGroup](#implementation-without-projectgroup)
+    - [Resource sharing via hierarchical namespaces](#resource-sharing-via-hierarchical-namespaces)
+    - [Exclusively leveraging existing Taints and Tolerations](#exclusively-leveraging-existing-taints-and-tolerations)
+
+## Summary
+
+## Motivation
+
+Currently, all users of Gardener share all available seeds which each other. 
+As there is an interest to conveniently have the ability to schedule shoots on seeds with the ensurement that no other user groups will have shoot control planes on this seed, a new system or functionality needs to be introduced to systematically enable such a scenario.
+
+### Goals
+
+The overarching goal of this GEP is to lay some foundational work to enable homogeneous sovereign cloud scenarios.
+
+- Enable projects to use some seeds exclusively by adapting the shoot scheduling ([`SeedBinding`s](#seedbinding))
+- Share those, so called, seed bindings to different Projects that logically belong together in a concept called [`ProjectGroup`s](#projectgroup)
+- Introduce new role for project group administrators
+
+### Non-Goals
+
+- Let users exclusively bind seeds on their own
+## Proposal
+
+#### ProjectGroup
+
+For that, a ProjectGroup API will be introduced that abstracts certain information over one or more projects. With this, it will become possible to homogeneously share resources between projects.
+
+Such a ProjectGroup manifest looks like this:
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: ProjectGroup
+metadata:
+  name: group1
+spec:
+  namespace: group1-projectgroup
+  projects:
+    - Project1
+    - Project2
+```
+
+The main idea behind `ProjectGroup`s is that they, like projects, have an own namespace.
+Every resource that is shareable (at first only `SeedBinding`s), which is in this namespace, will be shared to all `Project`s in this ProjectGroup.
+
+This sharing mechanism will be achieved by copying the objects from the project group namespace to the namespaces of the projects that belong to the project group.
```

Comment:
In my opinion, copying is better than referencing, as this would require another type of resource (SeedBindingReference or SeedBindingBinding) and could confuse the user. But I might also be very wrong about that. What do you think?

Reply:
Replication has some costs associated with it, e.g. additional runtime and storage costs. In addition to that, it increases complexity as project members might be tempted to change the copy instead of the source object.
We copy a lot of resources in the Gardener space, but references are also used. Therefore, both is possible.
I am interested if the trade-offs were considered, why we chose the way it is and how we want to address the short-comings of the approach.